flink-dist/src/main/assemblies/opt.xml

```diff
@@ -118,13 +118,28 @@
 			<fileMode>0644</fileMode>
 		</file>
 
-		<file>
+        <file>
 			<source>target/temporary/flink-shaded-zookeeper-3-${zookeeper.optional.version}-${flink.shaded.version}.jar</source>
 			<outputDirectory>opt</outputDirectory>
 			<destName>flink-shaded-zookeeper-${zookeeper.optional.version}.jar</destName>
 			<fileMode>0644</fileMode>
 		</file>
-	</files>
+
+        <file>
+            <source>../flink-dstl/flink-dstl-dfs/target/flink-dstl-dfs-${project.version}.jar</source>
```

Comment:
you should add dependencies to flink-dist for both of these modules to ensure they are built before flink-dist.